pkg/apis/numaflow/v1alpha1/pipeline_types.go

```diff
@@ -696,6 +708,23 @@ func (pls *PipelineStatus) SetObservedGeneration(value int64) {
 	pls.ObservedGeneration = value
 }
 
+// IsHealthy indicates whether the pipeline is in healthy status
+func (pls *PipelineStatus) IsHealthy() bool {
+	switch pls.Phase {
+	case PipelinePhaseFailed:
+		return false
+	case PipelinePhaseRunning:
+		return pls.IsReady()
```

Comment:
`IsReady` checks if all the `Conditions` are true, which are supposed to include conditions to track children status. @chandankumar4